Objects/dictobject.c

```diff
@@ -1467,6 +1467,17 @@ PyDict_GetItemWithError(PyObject *op, PyObject *key)
     return value;
 }
 
+PyObject *
+_PyDict_GetItemWithError(PyObject *dp, PyObject *kv)
+{
+    assert(PyUnicode_CheckExact(kv));
```

Comment:
Just a half-baked idea: would it be possible to initialize all hashes of global strings at interpreter startup, then eliminate some branching with a special 
```c
int
_PyDict_GetItemStrWithHashInitialized(PyDictObject *mp, PyUnicodeObject *key)
{
    assert(PyDict_CheckExact(mp));
    assert(PyUnicode_CheckExact(key));
    Py_hash_t hash = ((PyASCIIObject *)key)->hash;
    assert(hash != -1);
    // Inline _PyDict_GetItem_KnownHash --> _Py_dict_lookup
    PyDictKeysObject *dk = mp->ma_keys;
    Py_ssize_t ix = dictkeys_stringlookup(dk, key, hash);
    if (ix == DKIX_EMPTY) {
        return NULL;
    }
    else if (kind == DICT_KEYS_SPLIT) {
        return mp->ma_values->values[ix];
    }
    else {
        return DK_ENTRIES(dk)[ix].me_value;
    }
}
```